### PR TITLE
WIP: apiserver-encryption: use listers in deployers instead of client

### PR DESCRIPTION
**Pull request by sttts**
We did consistent client reads in the apiserver encryption deployers while eventual consistency with a cache (lister) is good enough.

### CODE REVIEW

File: pkg/operator/encryption/deployer/revisionedpod.go [hunk 1]
```diff
@@ -89,6 +80,10 @@ func NewRevisionLabelPodDeployer(
 // DeployedEncryptionConfigSecret returns the deployed encryption config and whether all
 // instances of the operand have acknowledged it.
 func (d *RevisionLabelPodDeployer) DeployedEncryptionConfigSecret() (secret *corev1.Secret, converged bool, err error) {
+	if !d.HasSynced() {
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/deployer/revisionedpod.go:83]: why do we need an informer here?
why we cannot have two listers instead?

usually, controllers are responsible for checking the caches have been sync before acting

File: pkg/operator/encryption/deployer/revisionedpod.go [hunk 2]
```diff
@@ -97,21 +92,23 @@ func (d *RevisionLabelPodDeployer) DeployedEncryptionConfigSecret() (secret *cor
 		return nil, false, nil
 	}
 
-	// do a live list so we never get confused about what revision we are on
-	apiServerPods, err := d.podClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "apiserver=true"})
+	// do a lister list. This can lag, but a workload controller like deployment can lag too.
+	apiserverPodSelector, _ := labels.Parse("apiserver=true")
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/deployer/revisionedpod.go:96]: check the error

File: pkg/operator/v1helpers/informers.go [hunk 3]
```diff
@@ -48,6 +49,17 @@ func (i kubeInformersForNamespaces) Start(stopCh <-chan struct{}) {
 	}
 }
 
+func (i kubeInformersForNamespaces) WaitForCacheSync(stopCh <-chan struct{}) bool {
```
Comment (p0lyn0mial) [inline pkg/operator/v1helpers/informers.go:52]: where is that method used?

File: pkg/operator/encryption/deployer/revisionedpod.go (hunk 2 above)
Comment (sttts) [reply pkg/operator/encryption/deployer/revisionedpod.go:96]: done

File: pkg/operator/encryption/deployer/revisionedpod.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/encryption/deployer/revisionedpod.go:83]: I don't like to fall over an non-synced informer here, reporting `converged=true` by accident. Better be defensive.

File: pkg/operator/v1helpers/informers.go (hunk 3 above)
Comment (sttts) [reply pkg/operator/v1helpers/informers.go:52]: in tests

**Pull request closed without merging**